grpc/interop/src/main/scala/grpc/testing/ServerMain.scala

```diff
@@ -0,0 +1,92 @@
+package grpc.testing
+
+import com.google.protobuf.CodedOutputStream
+import com.twitter.conversions.time._
+import com.twitter.finagle.buoyant.H2
+import com.twitter.io.Buf
+import com.twitter.util.{Await, Future, Promise, Throw}
+import io.buoyant.grpc.runtime.{Stream, ServerDispatcher}
+import java.nio.ByteBuffer
+import java.util.Arrays
+
+/**
+ * The main object for running a gRPC interop server.
+ *
+ * The interop tests are described here:
+ *   https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md
+ */
+object ServerMain {
+  def main(args: Array[String]) {
+
+    val iface = new TestService {
+      def emptyCall(empty: Empty): Future[Empty] = {
+        Future.value(new Empty())
+      }
+
+      def unaryCall(req: SimpleRequest): Future[SimpleResponse] = {
+        Future.value(new SimpleResponse(None, None, None))
+      }
+
+      def cacheableUnaryCall(req: SimpleRequest): Future[SimpleResponse] = {
+        Future.value(new SimpleResponse(None, None, None))
+      }
+
+      /**
+       * Echo back each request with a Payload having the requested size
+       */
+      def fullDuplexCall(req: Stream[StreamingOutputCallRequest]): Stream[StreamingOutputCallResponse] = {
+        val tx = Stream[StreamingOutputCallResponse]
+
+        req.recv().map { outputCallRequest =>
+          outputCallRequest.value.payload.map { payload =>
+            payload.body.map { body =>
+              val chars = Array.fill(body.length) { 0.toByte } // {} specializes us to a Byte array
+              val toSend = Buf.ByteArray.Owned(chars)
+              tx.send(StreamingOutputCallResponse(Some(Payload(None, Some(toSend)))))
+            }
+          }
+        }
+
+        return tx
+      }
+
+      // TODO: if an interop test can be found that needs this, we will implement it.
+      def halfDuplexCall(req: Stream[StreamingOutputCallRequest]): Stream[StreamingOutputCallResponse] = ???
+
+      /**
+       * Returns the aggregated size of input payloads.
+       */
+      def streamingInputCall(req: Stream[StreamingInputCallRequest]): Future[StreamingInputCallResponse] = {
+        var size = 0
+
+        req.recv().map { _.value.payload.map { _.body.map { size += _.length } } }
+
+        Future.value(StreamingInputCallResponse(Some(size)))
+      }
```

Comment:
Unfortunately, `client_streaming` is still broken with this but only after a certain size is hit so I suspect it's related to the issue you mentioned in #980.